### PR TITLE
Rename category -> itemGroup

### PR DESCRIPTION
**Pull request by svr333**
## Description
This is a follow up to #3139, a lot of comments and variables and error messages needed to be updated still.
This is what that does

Two more follow up questions:
- in `wiki.json`, `Christmas-Seasonal-Category` is mentioned, change to `Christmas-Seasonal-ItemGroup`?
- do I update `messages.yml` (and other files) keys? (eg: `guide.tooltips.open-category` -> `guide.tooltips.open-itemgroup`)

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java [hunk 1]
```diff
@@ -424,10 +424,10 @@ protected void openScriptDownloader(Player p, Block b, int page) {
         });
 
         int index = 0;
-        int categoryIndex = 45 * (page - 1);
+        int itemGroupIndex = 45 * (page - 1);
 
         for (int i = 0; i < 45; i++) {
-            int target = categoryIndex + i;
+            int target = itemGroupIndex + i;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:430]: This does not refer to item groups, it's the different categories for android scripts

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks good, just one small mistake.

Thanks!

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java (hunk 1 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:430]: ah all right

**Comment by svr333** [comment]
Also what about these
> - in `wiki.json`, `Christmas-Seasonal-Category` is mentioned, change to `Christmas-Seasonal-ItemGroup`?
> - do I update `messages.yml` (and other files) keys? (eg: `guide.tooltips.open-category` -> `guide.tooltips.open-itemgroup`)

**Comment by TheBusyBiscuit** [comment]
> Also what about these @TheBusyBiscuit
> 
> > * in `wiki.json`, `Christmas-Seasonal-Category` is mentioned, change to `Christmas-Seasonal-ItemGroup`?
> > * do I update `messages.yml` (and other files) keys? (eg: `guide.tooltips.open-category` -> `guide.tooltips.open-itemgroup`)

Regarding the wiki reference, the wiki page itself would need to be renamed first (if thats wanted).
Message strings... well, they *could* be updated but they would need to be updated in every class and every locale file. If you wanna go through all of that, sure. Otherwise we can just leave it as is or do it later.

**Comment by svr333** [comment]
I think I'm going to keep the Christmas seasonal category as-is, because I think the in-game text also still says category over itemgroup, and in this case it does make more sense  
its a christmas category

Ill do the other changes in a bit

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by WalshyDev — APPROVED** [review]
God GitHub hates large PRs but I couldn't find any typos or anything so looks good

**Pull request merged**